sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -456,7 +457,8 @@ def run_pipeline(self, pipeline, options):
       self._default_environment = (
           environments.DockerEnvironment.from_container_image(
               apiclient.get_container_image_from_options(options),
-              artifacts=environments.python_sdk_dependencies(options)))
+              artifacts=environments.python_sdk_dependencies(options),
+              resource_hints=environments.resource_hints_from_options(options)))
```

Comment:
I thought about it and my conclusion was that it is safer to add options in both places since:
- Hints are stored in Envrionment
- Environments can be created from Pipeline options and the latter can supply hints.
- In the future we plan to let users create Envrionments explicitly.
- Not sure whether current interactive runner will always pay attention to the _root_transform() since it executes/replaces parts of the pipeline. 

I did add https://github.com/apache/beam/pull/14390/files#diff-8fbc12ce22548ada33496537fd5ee055d7d6d5c64e10080ece3448e4ae1f6d26R223 though.